service/order.go

```diff
@@ -273,6 +319,25 @@ func (s *OrderService) Delete(ctx context.Context, in *pb.OrderSpecificRequest)
 		return nil, errors.E(errors.Op("Delete order"), err)
 	}
 
+	order := &pb.Order{}
+	err = proto.Unmarshal(orderInBytes, order)
+	if !errors.IsEmpty(err) {
+		return &pb.Empty{}, errors.E(errors.Op("Unmarshal order proto in Delete"), err)
+	}
+
+	_, publickey, err := identity.GetIdentity(s.Storage)
+	if !errors.IsEmpty(err) {
+		return &pb.Empty{}, errors.E(errors.Op("Get public key in Delete"), err)
+	}
+
+	isCreator, err := s.VerifyOrder(publickey, order)
+	if !errors.IsEmpty(err) {
+		return &pb.Empty{}, errors.E(errors.Op("Verify the order"), err)
+	}
+	if !isCreator {
```

Comment:
We don't want to restrict people from deleting orders from their own machine. Remove the return and wrap the following `s.P2p.Send()` in an `if isCreator` instead.